modules/dataplex-autodq/variables.tf

```diff
@@ -0,0 +1,177 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+variable "data" {
+  description = "The data source for DataScan. The input variable 'data' should be an object containing a single key-value pair that can be one of: * `entity`: The Dataplex entity that represents the data source (e.g. BigQuery table) for DataScan, of the form: `projects/{project_number}/locations/{locationId}/lakes/{lakeId}/zones/{zoneId}/entities/{entityId}`. * `resource`: The service-qualified full resource name of the cloud resource for a DataScan job to scan against. The field could be: BigQuery table of type 'TABLE' for DataProfileScan/DataQualityScan format, e.g: `//bigquery.googleapis.com/projects/PROJECT_ID/datasets/DATASET_ID/tables/TABLE_ID`."
+  type = object({
+    entity   = optional(string)
+    resource = optional(string)
+  })
+  validation {
+    condition     = length([for k, v in var.data : v if contains(["resource", "entity"], k) && v != null]) == 1
+    error_message = "Datascan data must specify one of 'entity', 'resource'."
+  }
+}
+
+variable "execution_trigger" {
+  description = "The input variable 'execution_trigger' specifies when a scan should be triggered. If not specified, the default is `on_demand`, which means the scan will not run until the user calls `dataScans.run` API. Alternatively, you can set `execution_trigger` to `scheduled` scheduled to run periodically based on a cron schedule expression."
+  type = object({
+    on_demand = optional(object({}))
```

Comment:
this could also be done differently: the variable is `execution_schedule` with string type, and defaults to `null` which implies on demand.

Reply:
Yes this is a good point. I didn't realize the default behaviour for the official API was on_demand until after I've done everything else. I will refactor this according to your suggestion